File: apps/backend/app/schemas/company.py
from pydantic import BaseModel, EmailStr, Field


class CompanySummary(BaseModel):
    id: str
    legal_name: str
    role: str


class CompanyOnboardingRequest(BaseModel):
    legal_name: str = Field(min_length=1, max_length=160)
    trade_name: str | None = Field(default=None, max_length=160)


class CompanyInformation(BaseModel):
    id: str
    legal_name: str
    trade_name: str | None = None
    billing_email: EmailStr | None = None
    address_line1: str | None = None
    address_line2: str | None = None
    city: str | None = None
    state: str | None = None
    postal_code: str | None = None
    invoice_prefix: str
    payment_terms_label: str
    payment_terms_days: int


class CompanyUpdateRequest(BaseModel):
    legal_name: str = Field(min_length=1, max_length=160)
    trade_name: str | None = Field(default=None, max_length=160)
    billing_email: EmailStr | None = None
    address_line1: str | None = Field(default=None, max_length=160)
    address_line2: str | None = Field(default=None, max_length=160)
    city: str | None = Field(default=None, max_length=120)
    state: str | None = Field(default=None, max_length=120)
    postal_code: str | None = Field(default=None, max_length=40)
    invoice_prefix: str = Field(min_length=1, max_length=20)
    payment_terms_label: str = Field(min_length=1, max_length=80)
    payment_terms_days: int = Field(ge=0, le=180)


class CompanyMemberRead(BaseModel):
    id: str
    full_name: str
    email: EmailStr
    role: str

File: apps/frontend/lib/company.ts
import { apiFetch } from "@/lib/api";

export type CompanyInformation = {
  id: string;
  legal_name: string;
  trade_name: string | null;
  billing_email: string | null;
  phone_number: string | null;
  website: string | null;
  dot_number: string | null;
  mc_number: string | null;
  ein_number: string | null;
  address_line1: string | null;
  address_line2: string | null;
  city: string | null;
  state: string | null;
  postal_code: string | null;
  remittance_name: string | null;
  remittance_address_line1: string | null;
  remittance_address_line2: string | null;
  remittance_city: string | null;
  remittance_state: string | null;
  remittance_postal_code: string | null;
  invoice_prefix: string;
  payment_terms_label: string;
  payment_terms_days: number;
};

export type CompanyMember = {
  id: string;
  full_name: string;
  email: string;
  role: string;
};

export type CompanyUpdateValues = Omit<CompanyInformation, "id">;

export function getCurrentCompany(cookieHeader?: string) {
  return apiFetch<CompanyInformation>("/companies/current", {
    method: "GET",
    headers: cookieHeader ? { Cookie: cookieHeader } : undefined
  });
}

export function updateCurrentCompany(values: CompanyUpdateValues) {
  return apiFetch<CompanyInformation>("/companies/current", {
    method: "PATCH",
    body: JSON.stringify(values)
  });
}

export function getCompanyMembers(cookieHeader?: string) {
  return apiFetch<CompanyMember[]>("/companies/current/members", {
    method: "GET",
    headers: cookieHeader ? { Cookie: cookieHeader } : undefined
  });
}
